cdap-proto/src/main/java/io/cdap/cdap/proto/security/Credential.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.proto.security;
+
+/**
+ * Encapsulating class for credentials passed in through CDAP.
+ */
+public class Credential {
+  public static final String CREDENTIAL_TYPE_BEARER = "Bearer";
+  public static final String CREDENTIAL_TYPE_INTERNAL = "CDAP-Internal";
+  public static final String CREDENTIAL_TYPE_EXTERNAL = "CDAP-External";
+  public static final String CREDENTIAL_TYPE_EXTERNAL_ENCRYPTED = "CDAP-External-Encrypted";
+
+  /**
+   * Identifies the type of credential.
+   */
+  public enum CredentialType {
```

Comment:
Could you add some docs on which one is which? What is the difference between External and Bearer?

Reply:
Fixed, Bearer was a relic from when we set the Credential class from external tokens too. However, I've removed it now that we no longer use it since Bearer types are all converted to External in the router.